packages/@react-spectrum/picker/test/Picker.test.js

```diff
@@ -56,114 +63,1398 @@ describe('Picker', function () {
+
+      let picker = getByRole('button');
+      act(() => {fireEvent.keyDown(picker, {key: 'ArrowDown'});});
+      act(() => jest.runAllTimers());
+
+      let listbox = getAllByRole('listbox')[1]; // ignore the one in the background for now
+      expect(listbox).toBeVisible();
+      expect(onOpenChange).toBeCalledTimes(1);
+      expect(onOpenChange).toHaveBeenCalledWith(true);
+      expect(picker).toHaveAttribute('aria-expanded', 'true');
+      expect(picker).toHaveAttribute('aria-controls', listbox.id);
+
+      let items = within(listbox).getAllByRole('option');
+      expect(items.length).toBe(3);
+      expect(items[0]).toHaveTextContent('One');
+      expect(items[1]).toHaveTextContent('Two');
+      expect(items[2]).toHaveTextContent('Three');
+
+      expect(document.activeElement).toBe(items[0]);
+    });
+
+    it('can be opened on ArrowUp key down and auto focuses the last item', function () {
+      let onOpenChange = jest.fn();
+      let {getAllByRole, getByRole} = render(
+        <Provider theme={theme}>
+          <Picker label="Test" onOpenChange={onOpenChange}>
+            <Item>One</Item>
+            <Item>Two</Item>
+            <Item>Three</Item>
+          </Picker>
+        </Provider>
+      );
+
+      expect(getAllByRole('listbox')).toHaveLength(1);
+
+      let picker = getByRole('button');
+      act(() => {fireEvent.keyDown(picker, {key: 'ArrowUp'});});
+      act(() => jest.runAllTimers());
+
+      let listbox = getAllByRole('listbox')[1]; // ignore the one in the background for now
+      expect(listbox).toBeVisible();
+      expect(onOpenChange).toBeCalledTimes(1);
+      expect(onOpenChange).toHaveBeenCalledWith(true);
+      expect(picker).toHaveAttribute('aria-expanded', 'true');
+      expect(picker).toHaveAttribute('aria-controls', listbox.id);
+
+      let items = within(listbox).getAllByRole('option');
+      expect(items.length).toBe(3);
+      expect(items[0]).toHaveTextContent('One');
+      expect(items[1]).toHaveTextContent('Two');
+      expect(items[2]).toHaveTextContent('Three');
+
+      expect(document.activeElement).toBe(items[2]);
+    });
+
+    it('supports controlled open state', function () {
+      let onOpenChange = jest.fn();
+      let {getAllByRole} = render(
+        <Provider theme={theme}>
+          <Picker label="Test" isOpen onOpenChange={onOpenChange}>
+            <Item>One</Item>
+            <Item>Two</Item>
+            <Item>Three</Item>
+          </Picker>
+        </Provider>
+      );
+
+      act(() => jest.runAllTimers());
+
+      expect(getAllByRole('listbox')).toHaveLength(2);
+      expect(onOpenChange).not.toBeCalled();
+
+      let picker = getAllByRole('button')[0];
+      expect(picker).toHaveAttribute('aria-expanded', 'true');
+
+      let listbox = getAllByRole('listbox')[1]; // ignore the one in the background for now
+      expect(listbox).toBeVisible();
+      expect(picker).toHaveAttribute('aria-controls', listbox.id);
```

Comment:
Why doesn't this have `aria-expanded` like the tests above?

Reply:
it does, see line 304